### PR TITLE
br:  disk space check follow up

### PR DESCRIPTION
**Pull request by RidRisR**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/54316, https://github.com/pingcap/tidb/issues/54470

Problem Summary:

### What changed and how does it work?

Now, the preserve rate of tiflash is set 1.3
Add more log and normalized errors
Fix the error that `TiFlashReplicas` is not set

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Now, the preserve rate of tiflash is set 1.3
Add more log and normalized errors
Fix the error that `TiFlashReplicas` is not se
```

### CODE REVIEW

File: br/pkg/task/restore.go [hunk 1]
```diff
@@ -1240,37 +1240,41 @@ func getStores(ctx context.Context, mgr *conn.Mgr) (stores *http.StoresInfo, err
...
 }
 
-func EstimateTikvUsage(files []*backuppb.File, maxReplica uint64, storeCnt int) uint64 {
+func EstimateTikvUsage(files []*backuppb.File, replicaCnt uint64, storeCnt uint64) uint64 {
 	if storeCnt == 0 {
 		return 0
 	}
-	var totalSize uint64 = 0
+	if replicaCnt > storeCnt {
+		replicaCnt = storeCnt
```
Comment (Tristan1900) [inline br/pkg/task/restore.go:1248]: @3pointer just to double check, do we have topology constraint that makes sure no more than 1 replica from the same resign placed on the same store?

File: br/pkg/task/restore.go [hunk 2]
```diff
@@ -1240,37 +1240,41 @@ func getStores(ctx context.Context, mgr *conn.Mgr) (stores *http.StoresInfo, err
 	return stores, nil
 }
 
-func EstimateTikvUsage(files []*backuppb.File, maxReplica uint64, storeCnt int) uint64 {
+func EstimateTikvUsage(files []*backuppb.File, replicaCnt uint64, storeCnt uint64) uint64 {
```
Comment (Tristan1900) [inline br/pkg/task/restore.go:1243]: nit, a more common name could be `replicationFactor` or `rf`

**Review by Tristan1900 — APPROVED** [review]
LGTM, thanks for the change!

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/task/restore.go [hunk 3]
```diff
@@ -92,17 +92,17 @@ const (
...
+	defaultPiTRBatchSize             = 16 * 1024 * 1024
+	defaultRestoreConcurrency        = 128
+	defaultPiTRConcurrency           = 16
+	defaultPDConcurrency             = 1
+	defaultStatsConcurrency          = 12
+	defaultBatchFlushInterval        = 16 * time.Second
+	defaultFlagDdlBatchSize          = 128
+	resetSpeedLimitRetryTimes        = 3
+	maxRestoreBatchSizeLimit         = 10240
+	pb                        uint64 = 1024 * 1024 * 1024 * 1024 * 1024
```
[the remark this answers is not in the document]
Comment (RidRisR) [reply br/pkg/task/restore.go]: fixed

File: br/pkg/task/restore.go (hunk 1 above)
Comment (3pointer) [reply br/pkg/task/restore.go:1247]: Yes, our raft implementation make this constraint

**Review by Leavrth — APPROVED** [review]

**Pull request merged**